src/VisualStudio/Core/Test/Progression/ProgressionTestHelpers.vb

```diff
@@ -36,10 +36,19 @@ Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.Progression
 
         Public Sub AssertSimplifiedGraphIs(graph As Graph, xml As XElement)
             Dim graphXml = graph.ToSimplifiedXDocument()
-            If Not XNode.DeepEquals(graphXml, xml) Then
+            If Not XNode.DeepEquals(graphXml.Root, xml) Then
                 ' They aren't equal, so therefore the text representations definitely aren't equal.
                 ' We'll Assert.Equal those, so that way xunit will show nice before/after text
-                Assert.Equal(xml.ToString(), graphXml.ToString())
+                'Assert.Equal(xml.ToString(), graphXml.ToString())
+
+                ' In an attempt to diagnose some flaky tests, the whole contents of both objects will be output
+                Throw New Exception($"Graph XML was not equal, check for out-of-order elements.
```

Comment:
I disagree.  This Xunit method is regularly used with large strings (think source before and after a transformation, or testing localized strings) and seeing the huge output makes parsing the log really difficult; it's usually sufficient to see just the failing location, but this is an instance where to properly debug a flaky test, we want to see everything.